### PR TITLE
feat: add git-cliff changelog generator

### PR DESCRIPTION
**Pull request by bill-mahoney**
**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [ ] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

### CODE REVIEW

File: .github/workflows/changelog.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
+name: Centralized EdgeX Changelog Generation
+
+on:
+  push:
+    branches:
+      - changelog-generator
+  schedule:
+    - cron: '0 0 * * 0' # Run every Sunday at 00:00
+  workflow_dispatch: # Allow manual triggering
```
Comment (lenny-intel) [inline .github/workflows/changelog.yml:9]: Will committers be able to manually trigger?

File: cliff.toml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  { message = "^refactor", group = "<!-- 3 -->♻ Code Refactoring" },
+  { message = "^doc", group = "<!-- 4 -->📖 Documentation" },
+  { message = "^build\\(deps\\): [bB]ump.*", skip = true },
+  { message = "^build", group = "<!-- 5 -->👷 Build" },
+  { message = "^chore", group = "<!-- 6 -->⛏️ Chore" },
+  { message = "^ci", group = "<!-- 7 -->🤖 Continuous Integration" },
+  { message = "^style", group = "<!-- 8 -->🎨 Styling" },
+  { message = "^test", group = "<!-- 9 -->🧪 Testing" },
+  { body = ".*security", group = "<!-- 10 -->🛡️ Security" },
+  { message = "^build", group = "<!-- 11 -->⚙️ Miscellaneous Tasks" },
```
Comment (lenny-intel) [inline cliff.toml:78]: This one should never get matched due to being duplicate of line 71 message.

**Review by lenny-intel — APPROVED** [review]
LGTM

File: .github/workflows/changelog.yml (hunk 1 above)
Comment (bill-mahoney) [reply .github/workflows/changelog.yml:9]: Correct

**Review by ernestojeda — APPROVED** [review]
💯

**Pull request merged**